// react-day-picker-themeable/lib/Caption.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = Caption;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _PropTypes = require('./PropTypes');

var _PropTypes2 = _interopRequireDefault(_PropTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Caption(_ref) {
  var date = _ref.date;
  var locale = _ref.locale;
  var localeUtils = _ref.localeUtils;
  var onClick = _ref.onClick;
  var styling = _ref.styling;

  return _react2.default.createElement(
    'div',
    _extends({}, styling('dayPickerCaption', date, locale), { onClick: onClick, role: 'heading' }),
    localeUtils.formatMonthTitle(date, locale)
  );
}

Caption.propTypes = {
  date: _react.PropTypes.instanceOf(Date),
  locale: _react.PropTypes.string,
  localeUtils: _PropTypes2.default.localeUtils,
  onClick: _react.PropTypes.func,
  styling: _react.PropTypes.func
};
