// react-day-picker-themeable/lib/createStylingFromTheme.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _reactBase16Styling = require('react-base16-styling');

var _default = require('./themes/default');

var _default2 = _interopRequireDefault(_default);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cn = function cn() {
  for (var _len = arguments.length, classNames = Array(_len), _key = 0; _key < _len; _key++) {
    classNames[_key] = arguments[_key];
  }

  return classNames.filter(function (c) {
    return c;
  }).join(' ');
};

var COLORS = {
  BACKGROUND: 'base00',
  NAVBAR_BACKGROUND: 'base01',
  HOVER_BACKGROUND: 'base02',
  WEEKDAY_TEXT: 'base03',
  TEXT: 'base07',
  DISABLED_TEXT: 'base03',
  DISABLED_BACKGROUND: 'base01',
  OUTSIDE_TEXT: 'base02',
  OUTSIDE_BACKGROUND: 'base00',
  TODAY_BACKGROUND: 'base00',
  TODAY_TEXT: 'base08',
  SELECTED_BACKGROUND: 'base0C',
  SELECTED_TEXT: 'base00'
};

function dayColors(base16Theme, modifiers) {
  if (modifiers.indexOf('outside') !== -1) {
    return {
      color: base16Theme[COLORS.OUTSIDE_TEXT],
      backgroundColor: base16Theme[COLORS.OUTSIDE_BACKGROUND]
    };
  } else if (modifiers.indexOf('disabled') !== -1) {
    return {
      color: base16Theme[COLORS.DISABLED_TEXT],
      backgroundColor: base16Theme[COLORS.DISABLED_BACKGROUND]
    };
  } else if (modifiers.indexOf('selected') !== -1) {
    return {
      color: base16Theme[COLORS.SELECTED_TEXT],
      backgroundColor: base16Theme[COLORS.SELECTED_BACKGROUND]
    };
  } else if (modifiers.indexOf('today') !== -1) {
    return {
      color: base16Theme[COLORS.TODAY_TEXT],
      backgroundColor: base16Theme[COLORS.TODAY_BACKGROUND]
    };
  }

  return {
    color: base16Theme[COLORS.TEXT],
    backgroundColor: base16Theme[COLORS.BACKGROUND]
  };
}

function getStylingFromBase16(base16Theme) {
  return {
    dayPicker: function dayPicker(_ref, locale, interactionDisabled) {
      var style = _ref.style;
      var className = _ref.className;
      return {
        className: cn('DayPicker', locale && 'DayPicker--' + locale, interactionDisabled && 'DayPicker--interactionDisabled', className),
        style: _extends({}, style, {
          color: base16Theme[COLORS.TEXT],
          backgroundColor: base16Theme[COLORS.BACKGROUND]
        })
      };
    },

    dayPickerNavBar: function dayPickerNavBar(_ref2) {
      var style = _ref2.style;
      var className = _ref2.className;
      return {
        className: cn('DayPicker-NavBar', className),
        style: _extends({}, style, {
          backgroundColor: base16Theme[COLORS.NAVBAR_BACKGROUND]
        })
      };
    },

    dayPickerMonth: function dayPickerMonth(_ref3) {
      var style = _ref3.style;
      var className = _ref3.className;
      return {
        className: cn('DayPicker-Month', className),
        style: style
      };
    },

    dayPickerMonthWrapper: function dayPickerMonthWrapper(_ref4) {
      var style = _ref4.style;
      var className = _ref4.className;
      return {
        className: cn('DayPicker-Body', className),
        style: style
      };
    },

    dayPickerWeek: function dayPickerWeek(_ref5) {
      var style = _ref5.style;
      var className = _ref5.className;
      return {
        className: cn('DayPicker-Week', className),
        style: style
      };
    },

    dayPickerNavButton: function dayPickerNavButton(_ref6, type) {
      var style = _ref6.style;
      var className = _ref6.className;
      return {
        className: cn('DayPicker-NavButton', type && 'DayPicker-NavButton--' + type, className),
        style: style
      };
    },

    dayPickerWeekday: function dayPickerWeekday(_ref7) {
      var style = _ref7.style;
      var className = _ref7.className;
      return {
        className: cn('DayPicker-Weekday', className),
        style: _extends({}, style, {
          color: base16Theme[COLORS.WEEKDAY_TEXT]
        })
      };
    },

    dayPickerWeekdays: function dayPickerWeekdays(_ref8) {
      var style = _ref8.style;
      var className = _ref8.className;
      return {
        className: cn('DayPicker-Weekdays', className),
        style: style
      };
    },

    dayPickerWeekdaysRow: function dayPickerWeekdaysRow(_ref9) {
      var style = _ref9.style;
      var className = _ref9.className;
      return {
        className: cn('DayPicker-WeekdaysRow', className),
        style: style
      };
    },

    dayPickerCaption: function dayPickerCaption(_ref10) {
      var style = _ref10.style;
      var className = _ref10.className;
      return {
        className: cn('DayPicker-Caption', className),
        style: style
      };
    },

    dayPickerDay: function dayPickerDay(_ref11, day, modifiers) {
      var style = _ref11.style;
      var className = _ref11.className;
      return {
        className: cn('DayPicker-Day', modifiers.map(function (modifier) {
          return 'DayPicker-Day--' + modifier;
        }).join(' '), className),
        style: _extends({}, style, dayColors(base16Theme, modifiers))
      };
    }
  };
}

exports.default = (0, _reactBase16Styling.createStyling)(getStylingFromBase16, {
  defaultBase16: _default2.default
});
