// react-day-picker-themeable/lib/Day.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = Day;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function handleEvent(handler, day, modifiers) {
  if (!handler) {
    return undefined;
  }
  var dayState = {};
  modifiers.forEach(function (modifier) {
    dayState[modifier] = true;
  });
  return function (e) {
    e.persist();
    handler(e, day, dayState);
  };
}
function Day(_ref) {
  var day = _ref.day;
  var tabIndex = _ref.tabIndex;
  var empty = _ref.empty;
  var modifiers = _ref.modifiers;
  var onMouseEnter = _ref.onMouseEnter;
  var onMouseLeave = _ref.onMouseLeave;
  var onClick = _ref.onClick;
  var onKeyDown = _ref.onKeyDown;
  var onTouchStart = _ref.onTouchStart;
  var onTouchEnd = _ref.onTouchEnd;
  var onFocus = _ref.onFocus;
  var ariaLabel = _ref.ariaLabel;
  var ariaDisabled = _ref.ariaDisabled;
  var ariaSelected = _ref.ariaSelected;
  var children = _ref.children;
  var styling = _ref.styling;
  var isHovered = _ref.isHovered;
  var isOutside = _ref.isOutside;

  if (empty) {
    return _react2.default.createElement('div', _extends({
      role: 'gridcell',
      'aria-disabled': true
    }, styling('dayPickerDay', day, modifiers, isHovered)));
  }
  return _react2.default.createElement(
    'div',
    _extends({}, styling('dayPickerDay', day, modifiers, isHovered), {
      tabIndex: tabIndex,
      role: 'gridcell',
      'aria-label': ariaLabel,
      'aria-disabled': ariaDisabled.toString(),
      'aria-selected': ariaSelected.toString(),
      onClick: handleEvent(onClick, day, modifiers),
      onKeyDown: handleEvent(onKeyDown, day, modifiers),
      onMouseEnter: handleEvent(onMouseEnter, day, modifiers),
      onMouseLeave: handleEvent(onMouseLeave, day, modifiers),
      onTouchEnd: handleEvent(onTouchEnd, day, modifiers),
      onTouchStart: handleEvent(onTouchStart, day, modifiers),
      onFocus: handleEvent(onFocus, day, modifiers),
      'data-day-type': isOutside ? 'outside' : 'inside'
    }),
    children
  );
}

Day.propTypes = {
  day: _react.PropTypes.instanceOf(Date).isRequired,
  children: _react.PropTypes.node.isRequired,
  styling: _react.PropTypes.func.isRequired,

  ariaDisabled: _react.PropTypes.bool,
  ariaLabel: _react.PropTypes.string,
  ariaSelected: _react.PropTypes.bool,
  empty: _react.PropTypes.bool,
  modifiers: _react.PropTypes.array,
  onClick: _react.PropTypes.func,
  onKeyDown: _react.PropTypes.func,
  onMouseEnter: _react.PropTypes.func,
  onMouseLeave: _react.PropTypes.func,
  onTouchEnd: _react.PropTypes.func,
  onTouchStart: _react.PropTypes.func,
  onFocus: _react.PropTypes.func,
  tabIndex: _react.PropTypes.number,
  isHovered: _react.PropTypes.bool,
  isOutside: _react.PropTypes.bool
};

Day.defaultProps = {
  modifiers: [],
  empty: false
};
