// react-day-picker-themeable/lib/Month.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = Month;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _PropTypes = require('./PropTypes');

var _PropTypes2 = _interopRequireDefault(_PropTypes);

var _Weekdays = require('./Weekdays');

var _Weekdays2 = _interopRequireDefault(_Weekdays);

var _Helpers = require('./Helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Month(_ref) {
  var month = _ref.month;
  var locale = _ref.locale;
  var localeUtils = _ref.localeUtils;
  var captionElement = _ref.captionElement;
  var onCaptionClick = _ref.onCaptionClick;
  var children = _ref.children;
  var firstDayOfWeek = _ref.firstDayOfWeek;
  var styling = _ref.styling;
  var weekdayComponent = _ref.weekdayComponent;
  var weekdayElement = _ref.weekdayElement;
  var fixedWeeks = _ref.fixedWeeks;

  var captionProps = _extends({
    styling: styling,
    date: month,
    localeUtils: localeUtils,
    locale: locale,
    onClick: onCaptionClick ? function (e) {
      return onCaptionClick(e, month);
    } : undefined
  }, styling('dayPickerCaption', month, locale));
  var weeks = (0, _Helpers.getWeekArray)(month, firstDayOfWeek, fixedWeeks);
  return _react2.default.createElement(
    'div',
    styling('dayPickerMonth', locale),
    _react2.default.cloneElement(captionElement, captionProps),
    _react2.default.createElement(_Weekdays2.default, {
      styling: styling,
      locale: locale,
      localeUtils: localeUtils,
      weekdayComponent: weekdayComponent,
      weekdayElement: weekdayElement
    }),
    _react2.default.createElement(
      'div',
      _extends({}, styling('dayPickerMonthWrapper'), { role: 'grid' }),
      weeks.map(function (week, j) {
        return _react2.default.createElement(
          'div',
          _extends({ key: j }, styling('dayPickerWeek', j, locale), { role: 'gridcell' }),
          week.map(function (day) {
            return children(day, month, styling);
          })
        );
      })
    )
  );
}

Month.propTypes = {
  month: _react.PropTypes.instanceOf(Date).isRequired,
  captionElement: _react.PropTypes.node.isRequired,
  firstDayOfWeek: _react.PropTypes.number.isRequired,
  locale: _react.PropTypes.string.isRequired,
  localeUtils: _PropTypes2.default.localeUtils.isRequired,
  onCaptionClick: _react.PropTypes.func,
  children: _react.PropTypes.func.isRequired,
  styling: _react.PropTypes.func.isRequired,
  weekdayComponent: _react.PropTypes.func,
  weekdayElement: _react.PropTypes.element,
  fixedWeeks: _react.PropTypes.bool
};
